### PR TITLE
Updated RESTful API Reference

### PR DESCRIPTION
**Pull request by writinwaters**
### What problem does this PR solve?

_Briefly describe what this PR aims to solve. Include background context that will help reviewers understand the purpose of the PR._

### Type of change

- [x] Documentation Update

### CODE REVIEW

**Comment by writinwaters** [comment]
Validated and improved clarity of the API reference. I may have tempered with the intended meaning, please double check.

**Comment by writinwaters** [comment]
There is still room for improvement. The API descriptions are still on the thin side, but I think the current version is ready for publication.

**Review by guoyuhao2330 — APPROVED** [review]
LGTM

**Review by KevinHuSh — APPROVED** [review]

File: docs/references/api.md [hunk 1]
```diff
@@ -290,44 +335,36 @@ This will be called to get the answer to users' questions.
...
-|-----------|--------|----------|---------------------------------------------------------|
-| file      | file   | No       | Upload file.                                            |
-| kb_name   | string | No       | Choose the upload knowledge base name.                  |
-| parser_id | string | Yes      | Choose the parsing method.                              |
-| run       | string | Yes      | Parsing will start automatically when the value is "1". |
+### Parameter
+
+|   Name      | Type   | Required | Description                                             |
+|-------------|--------|----------|---------------------------------------------------------|
+| `file`      | file   | Yes      | The file to upload.                                     |
```
Comment (writinwaters) [inline docs/references/api.md:395]: ?

File: docs/references/api.md [hunk 2]
```diff
@@ -192,43 +234,46 @@ This should be called whenever there's new user coming to chat.
...
-| doc_ids | string | Yes | Document IDs which is delimited by comma, like `c790da40ea8911ee928e0242ac180005,c790da40ea8911ee928e0242ac180005`. The retrieved content is limited in these documents. |
+### Parameter
+
+|   Name           |  Type  | Required | Description   |
+|------------------|--------|----------|---------------|
+| `conversation_id`| string | Yes      | The ID of the conversation session. Call ['GET' /new_conversation](#create-conversation) to retrieve the ID.|
+| `messages`       |  json  | Yes      | The latest question in a JSON form, such as `[{"role": "user", "content": "How are you doing!"}]`|
+| `quote`          |  bool  |  No      | Default: true |
+| `stream`         |  bool  |  No      | Default: true |
+| `doc_ids`        | string |  No      | Document IDs delimited by comma, like `c790da40ea8911ee928e0242ac180005,c790da40ea8911ee928e0242ac180005`. The retrieved contents will be confined to these documents. |
```
Comment (writinwaters) [inline docs/references/api.md]: same doc ID

File: docs/references/api.md [hunk 3]
```diff
@@ -290,44 +335,36 @@ This will be called to get the answer to users' questions.
...
-| parser_id | string | Yes      | Choose the parsing method.                              |
-| run       | string | Yes      | Parsing will start automatically when the value is "1". |
+### Parameter
+
+|   Name      | Type   | Required | Description                                             |
+|-------------|--------|----------|---------------------------------------------------------|
+| `file`      | file   | Yes      | The file to upload.                                     |
+| `kb_name`   | string | Yes      | The name of the knowledge base to upload the file to.   |
+| `parser_id` | string |  No      | The parsing method (chunk template) to use.             |
+| `run`       | string |  No      | - 1: Automatically start file parsing.                  |
```
Comment (writinwaters) [inline docs/references/api.md]: What if I set `run` to -1 without setting `parser_id`?

File: docs/references/api.md [hunk 4]
```diff
@@ -290,44 +335,36 @@ This will be called to get the answer to users' questions.
...
-| kb_name   | string | No       | Choose the upload knowledge base name.                  |
-| parser_id | string | Yes      | Choose the parsing method.                              |
-| run       | string | Yes      | Parsing will start automatically when the value is "1". |
+### Parameter
+
+|   Name      | Type   | Required | Description                                             |
+|-------------|--------|----------|---------------------------------------------------------|
+| `file`      | file   | Yes      | The file to upload.                                     |
+| `kb_name`   | string | Yes      | The name of the knowledge base to upload the file to.   |
+| `parser_id` | string |  No      | The parsing method (chunk template) to use.             |
```
Comment (writinwaters) [inline docs/references/api.md]: Which values could I set?

File: docs/references/api.md (hunk 3 above)
Comment (KevinHuSh) [reply docs/references/api.md]: naive:General,
qa:Q&A
manual:Manual
table:Table
paper:Paper
book:Book
laws:Laws
presentation:Presentation
picture:Picture
one:One

File: docs/references/api.md (hunk 2 above)
Comment (KevinHuSh) [reply docs/references/api.md]: c790da40ea8911ee928e0242ac180005,23dsf34ree928e0242ac180005

**Comment by JinHai-CN** [comment]
Add DEMO link at the first paragraph of README

File: docs/references/api.md [hunk 5]
```diff
@@ -42,27 +51,60 @@ This should be called whenever there's new user coming to chat.
...
+
+- **message**: All conversations in the specified conversation session.
+    - role: `"user"` or `"assistant"`.
+    - content: The text content of user or assistant. The citations are in a format like `##0$$`. The number in the middle, 0 in this case, indicates which part in data.reference.chunks it refers to.
+    
+- **user_id**: This is set by the caller.
+- **reference**: Each reference corresponds to one of the assistant's answers in `data.message`.
+    - chunks
+        - `content_with_weight`: Content of the chunk.
+        - `docnm_kwd`: Name of the *hit* document.
```
Comment (JinHai-CN) [inline docs/references/api.md]: @KevinHuSh docname_kwd => doc_name ?

File: docs/references/api.md [hunk 6]
```diff
@@ -192,43 +234,46 @@ This should be called whenever there's new user coming to chat.
...
+| `stream`         |  bool  |  No      | Default: true |
+| `doc_ids`        | string |  No      | Document IDs delimited by comma, like `c790da40ea8911ee928e0242ac180005,c790da40ea8911ee928e0242ac180005`. The retrieved contents will be confined to these documents. |
 
 ### Response 
+
+- **answer**: The answer to the user's latest question.
+- **reference**: 
+    - chunks: The retrieved chunks that contribute to the answer.  
+        - content_with_weight: Content of the chunk.
+        - docnm_kwd: Name of the *hit* document.
```
Comment (JinHai-CN) [inline docs/references/api.md]: @KevinHuSh docnm_kwd => doc_name ?

File: docs/references/api.md [hunk 7]
```diff
@@ -444,22 +493,26 @@ Get document list based on the knowledge base name and corresponding parameters.
 }
 
 ```
+</details>
+## Delete documents 
+
+- Path: `/api/document/rm/`
+- Method: POST
```
Comment (JinHai-CN) [inline docs/references/api.md]: DELETE method ? @KevinHuSh

File: docs/references/api.md (hunk 5 above)
Comment (writinwaters) [reply docs/references/api.md]: I agree.

File: docs/references/api.md (hunk 7 above)
Comment (KevinHuSh) [reply docs/references/api.md]: Method: DELETE

File: docs/references/api.md [hunk 8]
```diff
@@ -443,28 +492,34 @@ Get document list based on the knowledge base name and corresponding parameters.
...
 }
 
-```
+</code></pre>
+</details>
+
+## Delete documents 
 
-## Delete document 
+- Path: `/api/document/`
```
Comment (JinHai-CN) [inline docs/references/api.md]: Remove slash from All API

File: docs/references/api.md [hunk 9]
```diff
@@ -443,28 +500,34 @@ Get document list based on the knowledge base name and corresponding parameters.
...
 }
 
-```
+</code></pre>
+</details>
+
+## Delete documents 
 
-## Delete document 
+- Path: `/api/document`
```
Comment (JinHai-CN) [inline docs/references/api.md]: Path is also not a very proper word. Change it to: Request URI

File: docs/references/api.md [hunk 10]
```diff
@@ -289,46 +359,50 @@ This will be called to get the answer to users' questions.
...
+
+### Request
+
+#### Request URI
+
+| Method   |        Request URI                                          |
+|----------|-------------------------------------------------------------|
+| GET      | `/api/document/get/<id>`                                    |
+
+### Response
```
Comment (KevinHuSh) [inline docs/references/api.md:377]: Binary file content.

**Review by JinHai-CN — APPROVED** [review]

**Pull request merged**